### PR TITLE
updates for webauthn support

### PR DESCRIPTION
**Pull request by fillup**
### Breaking Change
- Added parameter for $rpOrigin to mfaCreate, mfaList, and mfaVerify functions. This
  parameter is needed for WebAuthn MFA methods. 
- Updated to PHP 7.4 and added a lot of type declarations and other linter updates

### CODE REVIEW

**Comment by Schparky** [comment]
>  This parameter is needed for WebAuthn MFA methods.

Careful how you use the word "method" as it's also used for "recovery methods". Normally wouldn't be a concern, but "recovery methods" is often abbreviated as simply "method" so it gets tricky. I'm mostly making this comment for other readers (including my future self).

File: composer.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "keywords": ["idp", "id broker", "api", "php client"],
   "license": "MIT",
   "require": {
-    "php": ">=7.0",
+    "php": ">=7.4",
```
Comment (Schparky) [inline composer.json:8]: Wish I was more current on my PHP understanding, but are you aware of anything in this library that specifically requires version 7.4?

**Review by Schparky — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (fillup) [reply composer.json:8]: me too, but I think the syntax for typed parameters does

**Review by Baggerone — APPROVED** [review]

File: src/BaseClient.php [hunk 2]
```diff
@@ -68,8 +71,13 @@ private function getHttpClientFromConfig(array $config)
...
     }
-    
-    private function getDescriptionFromConfig(array $config)
+
+    /**
+     * @param array $config
+     * @return Description|mixed
+     * @throws \Exception
+     */
+    private function getDescriptionFromConfig(array $config): Description
```
Comment (forevermatt) [inline src/BaseClient.php:80]: This return type might cause you problems if the `@return Description|mixed` is actually accurate, as PHP will throw a type error if something other than a Description is returned with this return type in place.

**Review by forevermatt — APPROVED** [review]
Approved, with one precautionary warning

**Pull request merged**